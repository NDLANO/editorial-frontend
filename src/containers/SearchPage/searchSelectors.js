/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { createSelector } from 'reselect';

const getSearchFromState = state => state.search;

export const getResults = createSelector(
  [getSearchFromState],
  search => search.results,
);

export const getSearching = createSelector(
  [getSearchFromState],
  search => search.searching,
);

export const getLastPage = createSelector([getSearchFromState], search =>
  Math.ceil(search.totalCount / search.pageSize),
);
