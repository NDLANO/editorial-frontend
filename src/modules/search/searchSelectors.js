/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { createSelector } from 'reselect';

const getSearchFromState = state => state.search;

export const getResults = createSelector(
  [getSearchFromState],
  search => search.totalSearchResults,
);

export const getSearching = createSelector(
  [getSearchFromState],
  search => search.searching,
);

export const getTotalResultsCount = createSelector(
  [getSearchFromState],
  search => search.totalSearchResults.totalCount,
);

export const getLastPage = createSelector(
  [getSearchFromState, getTotalResultsCount],
  (search, totalResultsCount) => {
    const largestPageSize = search.totalSearchResults.pageSize;
    return totalResultsCount
      ? Math.ceil(totalResultsCount / largestPageSize)
      : 1;
  },
);

export const getAudioTotalResultsCount = createSelector(
  [getSearchFromState],
  search => search.totalAudioResults.totalCount,
);

export const getImageTotalResultsCount = createSelector(
  [getSearchFromState],
  search => search.totalImageResults.totalCount,
);

export const getAudioLastPage = createSelector(
  [getSearchFromState, getAudioTotalResultsCount],
  (search, totalResultsCount) => {
    const largestPageSize = search.totalAudioResults.pageSize;
    return totalResultsCount
      ? Math.ceil(totalResultsCount / largestPageSize)
      : 1;
  },
);

export const getImageLastPage = createSelector(
  [getSearchFromState, getImageTotalResultsCount],
  (search, totalResultsCount) => {
    const largestPageSize = search.totalImageResults.pageSize;
    return totalResultsCount
      ? Math.ceil(totalResultsCount / largestPageSize)
      : 1;
  },
);

export const getAudioResults = createSelector(
  [getSearchFromState],
  search => search.totalAudioResults,
);

export const getImageResults = createSelector(
  [getSearchFromState],
  search => search.totalImageResults,
);
