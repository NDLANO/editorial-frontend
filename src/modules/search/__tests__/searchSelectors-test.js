/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { getResults, getLastPage, getDraftResults, getDraftLastPage } from '../searchSelectors';
import { contentResults, mediaResults } from './_mockSearchResult';

const lastPageTestState = {
  search: {
    searching: false,
    totalSearchResults: {
      totalCount: 30,
      pageSize: 3,
      results: [],
    },
    totalMediaResults: {
      totalCount: 30,
      pageSize: 3,
      results: [],
    },
  },
};

const lastPageTestMediaState = {
  search: {
    searching: false,
    totalDraftResults: {
      results: [
        {
          totalCount: 3,
          pageSize: 3,
        },
        {
          totalCount: 30,
          pageSize: 3,
        },
      ],
    },
  },
};

test('searchSelectors getResults', () => {
  const state = {
    search: { totalSearchResults: contentResults },
  };
  expect(getResults(state)).toMatchSnapshot();
});

test('searchSelectors getLastPage', () => {
  expect(getLastPage(lastPageTestState)).toBe(10);
});

test('searchSelectors getDraftResults', () => {
  const state = {
    search: { totalDraftResults: { results: mediaResults } },
  };

  expect(getDraftResults(state)).toMatchSnapshot();
});

test('searchSelectors getDraftLastPage', () => {
  expect(getDraftLastPage(lastPageTestMediaState)).toBe(11);
});
