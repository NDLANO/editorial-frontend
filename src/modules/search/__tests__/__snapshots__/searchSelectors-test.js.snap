// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchSelectors getResults 1`] = `undefined`;
